pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -481,6 +482,7 @@ func (summer *shardSummer) shardGroup(ctx context.Context, expr *parser.Aggregat
 		Param:    expr.Param,
 		Grouping: expr.Grouping,
 		Without:  expr.Without,
+		PosRange: expr.PosRange,
```

Comment:
Note to reviewers: this and the other `PosRange` changes below are part of the fix for annotations having incorrect position information